src/main/java/com/questhelper/helpers/quests/legendsquest/LegendsQuest.java

```diff
@@ -305,7 +321,7 @@ public Map<Integer, QuestStep> loadSteps()
 		saveViy.addStep(inCaveRoom6, climbDownWinchHolyForce);
 		saveViy.addStep(inCaveRoom5, useSpellOnDoorHolyForce);
 		saveViy.addStep(inCaveRoom4, searchMarkedWallHolyForce);
-		saveViy.addStep(inCaveRoom3, enterGate2HolyForce);
+		saveViy.addStep(inCaveRoom3, enterGate2ToSource);
```

Comment:
This'll mean in the sidebar the section of 'Unlocking the source' will be highlighted rather than 'Option 2: Save'. I'd say personally I'd go for one of two options here to ensure appropriate sidebar highlighting:
1. Keep the existing step for saving Viyeldi, so the step in the sidebar remains.
2. Compress the 'returning' steps in the Save section in the sidebar to one ConditionalStep of say 'Return to the source through the caverns', which contains all of these 'goToXHolyForce' steps, and use that in the sidebar, so the overlay text contains specific actions and requirements but the sidebar is more compressed, and can make use of the same 'enterGate2ToSource' step (as the sidebar highlighting checks the children of an active step to detemine what to highlight)

Reply:
Good catch - this was leftover from me experimenting before our chat about `copy()`